src/sagemaker/jumpstart/cache.py

```diff
@@ -339,11 +339,13 @@ def _retrieval_function(
                 formatted_content=model_specs
             )
         if data_type == HubDataType.MODEL:
-            hub_name, region, model_name, model_version = utils.extract_info_from_hub_content_arn(
+            info = utils.get_info_from_hub_resource_arn(
                 id_info
             )
-            hub = CuratedHub(hub_name=hub_name, region=region)
-            hub_content = hub.describe_model(model_name=model_name, model_version=model_version)
+            hub = CuratedHub(hub_name=info.hub_name, region=info.region)
+            hub_content = hub.describe_model(
+                model_name=info.hub_content_name, model_version=info.hub_content_version
+            )
             utils.emit_logs_based_on_model_specs(
                 hub_content.content_document,
```

Comment:
This assumed parsed HubContent. Suggest either explicitly commenting that or waiting for that implementation.

Reply:
It's a feature branch so I feel comfortable merging with out any of the implementation done, but yes this will be parsed. The `HubContentDocument` should be parsed at this point too